### PR TITLE
feat(kubernetes): Add kubernetes input plugin

### LINKED ISSUE #1774: Input Plugin Request: Kubernetes
### Proposal:
Right now it is possible to gather some kubernetes metrics via the prometheus url within kubernetes api. However, we have found that the metrics we receive from that endpoint are not particularly useful because they are not in the correct unit. For example, they only report CPU usage as `usage in seconds`. 

Our current solution also has a scalability issue: Since we run the telegraf daemon in a daemonset (runs on every node of the cluster) we eventually knock over the API server because we are polling it too often.

### Desired behavior:
Instead we should poll the local kubelet api. It contains a `/stats` handler that contains the following useful urls:

* `/stats/summary`
* `/stats/container`

Both of these endpoints contain detailed information about the pods and containers running on a particular host. This gives us the advantage of being able to scale a kubernetes installation to hundreds of nodes each one running telegraf without killing the API server. 

### Use case: [Why is this important (helps with prioritizing requests)]
Right now Deis ships telegraf as part of our default monitoring solution. By getting this change into telegraf it would allow us to install deis as part of a kubernetes fleet of dozens or even hundreds of nodes without having change the telegraf configuration. 

I will begin work on this plugin fairly soon but I wanted to get this formality out of the way first =).

### PR DESCRIPTION
**Pull request by jchauncey**
### Required for all PRs:

- [x] CHANGELOG.md updated (we recommend not updating this until the PR has been approved by a maintainer)
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)
- [x] README.md updated (if adding a new plugin)

closes #1774

### CODE REVIEW

File: plugins/inputs/kubernetes/kubernetes.go [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+
+func buildPodMetrics(summaryMetrics *SummaryMetrics, acc telegraf.Accumulator) {
+	for _, pod := range summaryMetrics.Pods {
+		for _, container := range pod.Containers {
+			tags := map[string]string{
+				"node":      summaryMetrics.Node.NodeName,
+				"namespace": pod.PodRef.Namespace,
+				"name":      container.Name,
+				"pod":       pod.PodRef.Name,
+				"uid":       pod.PodRef.UID,
```
Comment (dgnorton) [inline plugins/inputs/kubernetes/kubernetes.go]: Will this and the other `uid` tags have very high or unlimited cardinality?

Comment (jchauncey) [inline plugins/inputs/kubernetes/kubernetes.go]: yes. everytime a pod dies and comes back it would have a new uuid. Same thing with pod name

**Comment by sparrc** [comment]
@jchauncey I'm concerned this plugin will generate stats with very high cardinality, it appears that you are storing a UUID as a tag for every:

- container
- pod
- volume

In a typical kubernetes setup, how frequently would you expect new UUIDs of each of these to get generated? how frequently in a high-volume kubernetes setup?

**Comment by jchauncey** [comment]
It largely depends. We could remove the uuid but we will still have high cardinality with the pod name because it has a unique identifier which changes every time the pod dies. 

Im guess over a month there may be a few hundred pod uuids created on a moderately sized cluster.

**Comment by sstarcher** [comment]
I can tell you we churn around 3-5k pods a day.

**Comment by jchauncey** [comment]
What I could do is have a flag that allows you to exclude pod name from the
metrics and remove uuid altogether

**Comment by sparrc** [comment]
@jchauncey AFAICT simply excluding the pod name & uuid will only lead to duplicate metrics overwriting each other. If flags are added they will need to completely disable to per-pod and per-container metrics from being gathered.

Does kubernetes not provide any sort of aggregated metrics? From what I can see in this PR the metrics are all very high cardinality.

**Comment by jchauncey** [comment]
No that's the point of this plugin is to get detailed pod metrics.

**Comment by jchauncey** [comment]
@sparrc @dgnorton Updated pr with the changes we discussed in slack

**Comment by jchauncey** [comment]
closes #1774

**Comment by rikatz** [comment]
Yeay!! This is a great plugin!!! :+1:

**Pull request merged**